sentry-ruby/lib/sentry-ruby.rb

```diff
@@ -189,11 +194,18 @@ def init(&block)
       @main_hub = hub
       @background_worker = Sentry::BackgroundWorker.new(config)
 
+      @session_flusher = if config.auto_session_tracking
+                           Sentry::SessionFlusher.new(config, client)
```

Comment:
@sl0thentr0py I think we should also check if the SDK is deactivated before initializing the flusher, especially to avoid the additional thread that won't be used. We can use `Configuration#enabled_in_current_env?` for it.